raiden-cli/src/raiden.ts

```diff
@@ -0,0 +1,213 @@
+      },
+      rpc: {
+        type: 'boolean',
+        default: true,
+        desc: 'Start with or without the RPC server',
+      },
+      rpccorsdomain: {
+        type: 'string',
+        default: 'http://localhost:*/*',
+        desc: 'Comma separated list of domains to accept cross origin requests.',
+      },
+      apiAddress: {
+        type: 'string',
+        default: '127.0.0.1:5001',
+        desc: 'host:port to bind to and listen for API requests',
+      },
+      routingMode: {
+        choices: ['pfs', 'local', 'private'] as const,
+        default: 'pfs',
+        desc: 'Anything else than "pfs" disables mediated transfers',
+      },
+      pathfindingServiceAddress: {
+        type: 'string',
+        default: 'auto',
+        desc: 'Force a given PFS to be used; "auto" selects cheapest registered on-chain',
+      },
+      enableMonitoring: {
+        type: 'boolean',
+        default: true,
+        desc: "By default, enables monitoring if there's a UDC deposit",
+      },
+    })
+    .env('RAIDEN')
+    .help().argv;
+}
+
+async function askUserForPassword(): Promise<string> {
+  const userInput = await inquirer.prompt<{ password: string }>([
+    { type: 'password', name: 'password', message: 'Private Key Password:', mask: '*' },
+  ]);
+  return userInput.password;
+}
+
+async function getWallet(
+  keystoreDir: string,
+  address: string,
+  passwordFile?: string,
+): Promise<Wallet> {
+  let password;
+  if (!passwordFile) password = await askUserForPassword();
+  else password = (await fs.readFile(passwordFile, 'utf-8')).split('\n').shift()!;
+  let jsonStr;
+  for (const filename of await fs.readdir(keystoreDir)) {
+    if (!(filename.startsWith('UTC-') || filename.endsWith('.json'))) continue;
+    try {
+      jsonStr = await fs.readFile(path.join(keystoreDir, filename), 'utf-8');
+      const jsonContent = JSON.parse(jsonStr);
+      const jsonAddr = jsonContent?.['address'];
+      if (!jsonAddr || !address.toLowerCase().endsWith(jsonAddr.toLowerCase())) throw '';
+      break;
+    } catch (e) {
+      jsonStr = undefined;
+      continue;
+    }
+  }
+  if (!jsonStr) throw new Error(`Could not find keystore file for "${address}"`);
+  return await Wallet.fromEncryptedJson(jsonStr, password);
+}
+
+function createLocalStorage(name: string): LocalStorage {
+  const localStorage = new LocalStorage(name);
+  Object.assign(globalThis, { localStorage });
+  return localStorage;
+}
+
+function shutdownServer(this: Cli): void {
+  if (this.server?.listening) {
+    this.log.info('Closing server...');
+    this.server.close();
+  }
+}
+
+function unrefTimeout(timeout: number | NodeJS.Timeout) {
+  if (typeof timeout === 'number') return;
+  timeout.unref();
+}
+
+function shutdownRaiden(this: Cli): void {
+  if (this.raiden.started) {
+    this.log.info('Stopping raiden...');
+    this.raiden.stop();
+    // force-exit at most 5s after stopping raiden
+    unrefTimeout(setTimeout(() => process.exit(0), 5000));
```

Comment:
Kinda.. we usually try our best to finish every timeout left, the SDK unsubscribes from everything, etc, but I don't have that much control over e.g. matrix-js-sdk, so itseems _sometimes_ some timeout/network handler hangs and prevents NodeJS from exiting after we stop the SDK. SP will complain on scenarios which expect the client to stop cleanly if it doesn't do so, and since any important request was already finished, I just give a little bit of time and force-exit. The `unref` is to prevent the `setTimeout` itself from hanging node (if everything else is already done), but only callable on Node's setTimeout (browser's is just `number`, that's why the typeguarded function was needed).